### PR TITLE
Evaluate measure support

### PR DESCRIPTION
**Pull request by mgramigna**
# Summary

Change the cli and calculation util to use `$evaluate-measure`. Syncing up with what we want the this to support for future integration into a CI tool.

Addresses 
* TACOSTRAT-397 - Use $evaluate-measure
* TACOSTRAT-379 - Include "no population" results

## New behavior

* Results for patients that didn't fall into any population are no written to disk
* Individual measure report is now written to `/measure-reports` for each patient

## Code changes

The whole dang thing

* Eliminate `--cql` arg, made `--measure-id` required
* Change csv to be of the form "bundle, population" instead of a bunch of booleans
* Episode of care counts and regular measure counts are treated the same (`> 0` in a population is considered to be "true")
* Use `fhirpath` to parse MeasureReport for relevant data
* Logs include timestamps
* Update unit tests/fixtures to adhere to $evaluate-measure behavior
* Version bump to `3.0.0`, gg on backwards compatibility
  * the `$cql` version of this will still be available for `2.X.X` and below
* Update docs

# Testing guidance

1) Start cqf ruler R4 and ensure that it is seeded with any number of [these bundles](https://github.com/DBCG/connectathon/tree/master/fhir4/bundles)
2) Pick a measure that's on the server, dealer's choice
3) You can use any test patients you want, might be easiest to just check the `patients-r4/numerator` and `patients-r4/denominator` directory for the corresponding measure that you choose in [fhir-patient-generator](https://github.com/projecttacoma/fhir-patient-generator)
4) Send it

``` bash
node src/cli.js -d <your-directory> -m <your-measure-id>
```

(url and period are defaulted but you can add those flags if you wish.

5) Behavior should be the same as before, writing patients to corresponding directories, logging output, failing on errors, etc. There should now be individual MeasureReports in the `/output/.../measure-reports` directory

### CODE REVIEW

File: test/calculation.test.js [hunk 1]
```diff
@@ -1,125 +1,87 @@
...
-  // Should have all true for the three populations
-  expect(result.initial_population).toBe(true);
-  expect(result.numerator).toBe(true);
-  expect(result.denominator).toBe(true);
+  expect(result.measureReport).toEqual(numerMeasureReport);
+  expect(result.population).toBe('numerator');
 });
 
-test('patient in ipp/denom but not numerator should yield proper booleans', async () => {
+test('patient in the denominator should yield an proper population result', async () => {
```
Comment (losborne) [inline test/calculation.test.js]:
```suggestion
test('patient in the denominator should yield a proper population result', async () => {
```

File: test/calculation.test.js [hunk 2]
```diff
@@ -1,125 +1,87 @@
...
-  // Should have all true for everything except numerator
-  expect(result.initial_population).toBe(true);
-  expect(result.numerator).toBe(false);
-  expect(result.denominator).toBe(true);
+  expect(result.measureReport).toEqual(denomMeasureReport);
+  expect(result.population).toBe('denominator');
 });
 
-test('illegal cql should yield no population results and an error message', async () => {
+test('patient in the ipop should yield an proper population result', async () => {
```
Comment (losborne) [inline test/calculation.test.js]:
```suggestion
test('patient in the ipop should yield a proper population result', async () => {
```

File: test/calculation.test.js [hunk 3]
```diff
@@ -1,125 +1,87 @@
...
     PERIOD_END,
   );
 
-  expect(mockExit).toHaveBeenCalledWith(1);
+  expect(result.measureReport).toEqual(ipopMeasureReport);
+  expect(result.population).toBe('ipop');
 });
 
-test('episode of care measure should contain episode counts', async () => {
+test('patient in no population should yield an proper population result', async () => {
```
Comment (losborne) [inline test/calculation.test.js]:
```suggestion
test('patient in no population should yield a proper population result', async () => {
```

**Comment by mgramigna** [comment]
Addressed suggestions and NITs

**Comment by mgramigna** [comment]
Added https://github.com/projecttacoma/fhir-patient-generator/pull/11. Updates the Makefiles to support fbc v3.0.0

File: package.json [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
   "author": "Matthew Gramigna <mgramigna@mitre.org>",
   "main": "src/cli.js",
   "scripts": {
-    "lint": "eslint ./**/*.js",
-    "lint-fix": "eslint ./**/*.js --fix",
-    "test": "NO_PROXY=localhost jest"
+    "lint": "eslint \"./**/*.js\"",
+    "lint-fix": "eslint \"./**/*.js\" --fix",
+    "test": "NO_PROXY=localhost jest",
+    "test:watch": "NO_PROXY=localhost jest --watchAll"
```
Comment (okeefm) [inline package.json]:
```suggestion
    "test": "jest",
    "test:watch": "jest --watchAll"
```

Shouldn't need the `NO_PROXY` var any more, I don't think?

**Comment by okeefm** [comment]
Looks good, just one minor change in `package.json`

**Comment by mgramigna** [comment]
Transparency baby!

**Comment by mgramigna** [comment]
Going to release and tag new version to npm, then merge

**Pull request merged**